testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -2005,10 +2135,105 @@ fn retry_on_timeout() {
     signer_test.shutdown();
 }
 
+#[test]
+#[ignore]
+/// This test checks that the signers will broadcast a block once they receive enough signatures.
+fn signers_broadcast_signed_blocks() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(EnvFilter::from_default_env())
+        .init();
+
+    info!("------------------------- Test Setup -------------------------");
+    let num_signers = 5;
+    let sender_sk = Secp256k1PrivateKey::new();
+    let sender_addr = tests::to_addr(&sender_sk);
+    let send_amt = 100;
+    let send_fee = 180;
+    let recipient = PrincipalData::from(StacksAddress::burn_address(false));
+    let mut signer_test: SignerTest<SpawnedSigner> = SignerTest::new(
+        num_signers,
+        vec![(sender_addr.clone(), send_amt + send_fee)],
+        Some(Duration::from_secs(5)),
+    );
+    let http_origin = format!("http://{}", &signer_test.running_nodes.conf.node.rpc_bind);
+
+    signer_test.boot_to_epoch_3();
+    sleep_ms(10_000);
+
+    signer_test.mine_nakamoto_block(Duration::from_secs(30));
+    sleep_ms(10_000);
+
+    TEST_IGNORE_SIGNERS.lock().unwrap().replace(true);
+
+    let blocks_before = signer_test
+        .running_nodes
+        .nakamoto_blocks_mined
+        .load(Ordering::SeqCst);
+
+    let signer_pushed_before = signer_test
+        .running_nodes
+        .nakamoto_blocks_signer_pushed
+        .load(Ordering::SeqCst);
+
+    let info_before = get_chain_info(&signer_test.running_nodes.conf);
+
+    // submit a tx so that the miner will mine a block
+    let sender_nonce = 0;
+    let transfer_tx =
+        make_stacks_transfer(&sender_sk, sender_nonce, send_fee, &recipient, send_amt);
+    submit_tx(&http_origin, &transfer_tx);
+
+    debug!("Transaction sent; waiting for block-mining");
+
+    let start = Instant::now();
+    let duration = 60;
+    loop {
```

Comment:
To cut down on how often this logic was getting repeated in tests, I added `wait_for()`, should use it here.